packages/amplication-client/src/Plugins/InstalledPluginSettings.tsx

```diff
@@ -0,0 +1,170 @@
+import { HorizontalRule, Snackbar } from "@amplication/design-system";
+import MonacoEditor, { Monaco } from "@monaco-editor/react";
+import classNames from "classnames";
+import * as monaco from "monaco-editor/esm/vs/editor/editor.api";
+import React, {
+  useCallback,
+  useContext,
+  useEffect,
+  useMemo,
+  useRef,
+  useState,
+} from "react";
+import { match } from "react-router-dom";
+import { BackNavigation } from "../Components/BackNavigation";
+import { Button, EnumButtonStyle } from "../Components/Button";
+import { AppContext } from "../context/appContext";
+import { setEditorTheme } from "../Resource/code-view/CodeViewEditor";
+import { AppRouteProps } from "../routes/routesUtil";
+import { formatError } from "../util/error";
+import usePlugins from "./hooks/usePlugins";
+import "./InstalledPluginSettings.scss";
+import { PluginLogo } from "./PluginLogo";
+
+type Props = AppRouteProps & {
+  match: match<{
+    resource: string;
+    plugin: string;
+  }>;
+};
+
+const InstalledPluginSettings: React.FC<Props> = ({
+  match,
+  moduleClass,
+}: Props) => {
+  const { resource, plugin: pluginInstallationId } = match.params;
+  const { currentProject, currentWorkspace } = useContext(AppContext);
+  const [isValid, setIsValid] = useState<boolean>(true);
+  const [initialValue, setInitialValue] = useState<any>("");
+  const [isDirty, setIsDirty] = useState<boolean>(false);
+
+  const {
+    pluginInstallation,
+    loadingPluginInstallation,
+    pluginCatalog,
+    updatePluginInstallation,
+    updateError,
+  } = usePlugins(resource, pluginInstallationId);
+
+  const editorRef = useRef<monaco.editor.IStandaloneCodeEditor | null>(null);
+
+  const plugin = useMemo(() => {
+    return (
+      pluginInstallation &&
+      pluginCatalog[pluginInstallation?.PluginInstallation.pluginId]
+    );
+  }, [pluginInstallation, pluginCatalog]);
+
+  function handleEditorDidMount(
+    editor: monaco.editor.IStandaloneCodeEditor,
+    monaco: Monaco
+  ) {
+    editorRef.current = editor;
+  }
+
+  function handleEditorChange(value: string | undefined) {
```

Comment:
I think this implementation of the editor is very problematic. First, the package itself is not updated to the latest react function component methodology and second, there are many states that will cause many renders (memory leaks) that can be avoid. third, it should be a stand a lone component that can be use anywhere in our application and also can be replaceable and decouple.

Reply:
I am not sure what you are suggesting, is there a newer version of monaco that works differently? 

About the encapsulation, I agree